Dockerfile

```diff
@@ -1,16 +1,18 @@
 FROM golang:alpine
-
+RUN mkdir -p /go/src/showhook/
+WORKDIR /go/src/showhook/
+ADD main.go .
 RUN apk --no-cache add git
-
-ADD main.go /go/src/app/main.go
-ADD public /go/src/app/public
-WORKDIR /go/src/app
 RUN go get -d -v
 RUN go install -v
+RUN go build main.go
 
+FROM alpine:3.7
+RUN mkdir -p /go/src/showhook/
+WORKDIR /go/src/showhook/
+COPY --from=0 /go/src/showhook/main .
```

Comment:
what does --from=0 do?

Reply:
This allows you to access another container's files.  This way you can have a **BUILD** container with build-tools and the golang compiler.  Meanwhile you create a second "FROM" **RUNTIME** container to host the binary at a fraction of the build container size.

Savings of 400MB down to 12MB.  🎉

Reply:
`FROM alpine:3.7` is `--from=0`

`FROM golang:alpine` is `--from=1`